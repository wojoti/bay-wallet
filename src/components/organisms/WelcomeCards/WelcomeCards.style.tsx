import {StyleSheet} from 'react-native';

interface Props {
  background?: string;
  radius?: number;
  width?: string;
  mt?: number;
  m?: number;
}
const style = ({background, radius, width, mt, m}: Props) =>
  StyleSheet.create({
    wrapper: {
      backgroundColor: background,
    },
    contentWithRadius: {
      backgroundColor: background,
      borderWidth: 0,
      borderColor: 'transparent',
      borderTopLeftRadius: radius,
      borderTopRightRadius: radius,
    },
    sizer: {
      width: width,
      marginTop: mt,
      marginHorizontal: m,
    },
    cardSizer: {
      width: '100%',
      alignItems: 'center',
    },
    pagination: {
      justifyContent: 'flex-start',
      left: 28,
      bottom: 80,
    },
  });
export default style;
