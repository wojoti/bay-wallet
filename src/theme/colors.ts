export const lightPrimary = '#4E439E';
export const primary = '#363062';
export const secondary = 'white';

export const darkGrey = '#595959';
export const grey = '#6B6B6B';
export const vlightGrey = '#D1D1D1';
export const borderGrey = '#E2E2E2';

export const iconGrey = '#BDBDBD';

export const searchGrey = '#F5F5F5';

export const separator = '#F6F6F6';

export const green = '#61C277';
export const lime = '#53E520';
export const red = '#E52020';

export const successGreen = '#3EB290';

export const pink = '#D8B9C3';
export const darkPurple = '#26204E';
export const balanceWhite = '#F8F8FF';

export const marketDarkGrey = '#3D454B';

export const cardYellow = '#FFEF60';

export const buttonGrey = '#EFEFEF';
