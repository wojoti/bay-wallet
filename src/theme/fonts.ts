export const font = 'Poppins-Light';
export const boldFont = 'Poppins-Bold';
