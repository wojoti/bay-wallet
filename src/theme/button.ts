import {StyleSheet} from 'react-native';
import {primary, secondary} from './colors';
const buttonStyle = StyleSheet.create({
  primary: {
    borderWidth: 0,
    backgroundColor: primary,
  },
  secondary: {
    borderColor: primary,
    backgroundColor: secondary,
  },
});

export const buttonTextStyle = StyleSheet.create({
  primary: {
    color: secondary,
  },
  secondary: {
    color: primary,
  },
});

export default buttonStyle;
