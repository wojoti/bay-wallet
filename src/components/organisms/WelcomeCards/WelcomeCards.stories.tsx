import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {primary, secondary} from 'theme/colors';

import WelcomeCards from './WelcomeCards';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Organisms/WelcomeCards',
  component: WelcomeCards,
  argTypes: {
    onButtonPress: {action: 'onButtonPress'},
  },
} as ComponentMeta<typeof WelcomeCards>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof WelcomeCards> = args => (
  <WelcomeCards {...args} />
);

export const Example = Template.bind({});
Example.args = {
  mainColor: primary,
  cardColor: secondary,
  mainText1: 'New cards in few clicks',
  cardText1:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.',
  mainText2: 'Easy payments',
  cardText2:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.',
  testId: '',
};
