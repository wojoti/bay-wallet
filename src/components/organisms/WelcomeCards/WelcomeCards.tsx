import {Container, Img, ImgType} from 'components/atoms';
import {WelcomeCardsArea} from 'components/molecules';
import {PropsWithChildren} from 'react';
import Swiper from 'react-native-swiper';
import {primary} from 'theme/colors';
import style from './WelcomeCards.style';

export type WelcomeCardsProps = PropsWithChildren<{
  onButtonPress: () => void;
  mainColor: string;
  cardColor: string;
  mainText1: string;
  cardText1: string;
  mainText2: string;
  cardText2: string;
  testId?: string;
}>;

const WelcomeCards = ({
  onButtonPress,
  mainColor,
  cardColor,
  mainText1,
  cardText1,
  mainText2,
  cardText2,
  testId,
}: WelcomeCardsProps) => {
  return (
    <Container
      flex={1}
      justify="center"
      align="center"
      customStyle={style({background: mainColor}).wrapper}
      testId={testId}>
      <Container
        flex={1.2}
        customStyle={style({background: mainColor}).wrapper}
        justify="flex-end">
        <Img
          src={ImgType.cardLock}
          height={330}
          alt={''}
          testId={'welcomecards-img'}
        />
      </Container>
      <Container
        flex={1}
        customStyle={
          style({background: cardColor, radius: 45}).contentWithRadius
        }
        align="center"
        testId={'welcomecards-radius'}>
        <Swiper
          loop={false}
          activeDotColor={primary}
          paginationStyle={style({}).pagination}>
          <WelcomeCardsArea
            onButtonPress={onButtonPress}
            mainText={mainText1}
            cardText={cardText1}
            testId={'welcomecards-welcomecardsarea1'}
          />

          <WelcomeCardsArea
            onButtonPress={onButtonPress}
            mainText={mainText2}
            cardText={cardText2}
            testId={'welcomecards-welcomecardsarea2'}
          />
        </Swiper>
      </Container>
    </Container>
  );
};

export default WelcomeCards;
