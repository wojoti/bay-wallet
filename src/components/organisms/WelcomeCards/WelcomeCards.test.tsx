import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import {primary, secondary} from 'theme/colors';
import WelcomeCards, {WelcomeCardsProps} from './WelcomeCards';

const handlePress = jest.fn();
const props: WelcomeCardsProps = {
  onButtonPress: handlePress,
  mainColor: primary,
  cardColor: secondary,
  mainText1: 'maintext1',
  cardText1: 'cardtext1',
  mainText2: 'maintext2',
  cardText2: 'cardtext2',
  testId: 'test-welcomecards-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<WelcomeCards {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render welcomecards', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
});

test('should render welcomecards with img', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsImgElement =
    within(welcomecardsElement).getByTestId('welcomecards-img');
  expect(welcomeCardsImgElement).toBeOnTheScreen();
});

test('should render welcomecards with 2 x welcomecardsarea ', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsWCAElement1 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea1',
  );
  expect(welcomeCardsWCAElement1).toBeOnTheScreen();
  const welcomeCardsWCAElement2 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea2',
  );
  expect(welcomeCardsWCAElement2).toBeOnTheScreen();
});

test('should render first welcomecardsarea: correct header ', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsWCAElement1 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea1',
  );
  expect(welcomeCardsWCAElement1).toBeOnTheScreen();
  const welcomecardsareaHeaderElement = within(
    welcomeCardsWCAElement1,
  ).getByTestId('welcomecardsarea-header');
  expect(welcomecardsareaHeaderElement).toBeOnTheScreen();
  const textElement = within(welcomecardsareaHeaderElement).getByText(
    'maintext1',
  );
  expect(textElement).toBeOnTheScreen();
});

test('should render first welcomecardsarea: correct label ', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsWCAElement1 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea1',
  );
  expect(welcomeCardsWCAElement1).toBeOnTheScreen();
  const welcomecardsareaLabelElement = within(
    welcomeCardsWCAElement1,
  ).getByTestId('welcomecardsarea-label');
  expect(welcomecardsareaLabelElement).toBeOnTheScreen();
  const textElement = within(welcomecardsareaLabelElement).getByText(
    'cardtext1',
  );
  expect(textElement).toBeOnTheScreen();
});

test('should render first welcomecardsarea: logo ', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsWCAElement1 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea1',
  );
  expect(welcomeCardsWCAElement1).toBeOnTheScreen();
  const welcomecardsareaImgElement = within(
    welcomeCardsWCAElement1,
  ).getByTestId('welcomecardsarea-logo');
  expect(welcomecardsareaImgElement).toBeOnTheScreen();
});

test('should render first welcomecardsarea: button and correct style', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsWCAElement1 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea1',
  );
  expect(welcomeCardsWCAElement1).toBeOnTheScreen();
  const welcomecardsareaButtonElement = within(
    welcomeCardsWCAElement1,
  ).getByTestId('welcomecardsarea-button');
  expect(welcomecardsareaButtonElement).toBeOnTheScreen();
  expect(welcomecardsareaButtonElement).toHaveStyle({
    backgroundColor: secondary,
  });
  const buttonText = within(welcomecardsareaButtonElement).getByTestId(
    'button-text',
  );
  expect(buttonText).toHaveStyle({
    color: primary,
  });
});

test('should render first welcomecardsarea: should handle onButtonPress ', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsWCAElement1 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea1',
  );
  expect(welcomeCardsWCAElement1).toBeOnTheScreen();
  const welcomecardsareaButtonElement = within(
    welcomeCardsWCAElement1,
  ).getByTestId('welcomecardsarea-button');
  expect(welcomecardsareaButtonElement).toBeOnTheScreen();
  expect(handlePress).toBeCalledTimes(0);
  fireEvent.press(welcomecardsareaButtonElement);
  expect(handlePress).toBeCalledTimes(1);
});

test('should render second welcomecardsarea: correct header ', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsWCAElement2 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea2',
  );
  expect(welcomeCardsWCAElement2).toBeOnTheScreen();
  const welcomecardsareaHeaderElement = within(
    welcomeCardsWCAElement2,
  ).getByTestId('welcomecardsarea-header');
  expect(welcomecardsareaHeaderElement).toBeOnTheScreen();
  const textElement = within(welcomecardsareaHeaderElement).getByText(
    'maintext2',
  );
  expect(textElement).toBeOnTheScreen();
});

test('should render second welcomecardsarea: correct label ', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsWCAElement2 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea2',
  );
  expect(welcomeCardsWCAElement2).toBeOnTheScreen();
  const welcomecardsareaLabelElement = within(
    welcomeCardsWCAElement2,
  ).getByTestId('welcomecardsarea-label');
  expect(welcomecardsareaLabelElement).toBeOnTheScreen();
  const textElement = within(welcomecardsareaLabelElement).getByText(
    'cardtext2',
  );
  expect(textElement).toBeOnTheScreen();
});

test('should render second welcomecardsarea: logo ', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsWCAElement2 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea2',
  );
  expect(welcomeCardsWCAElement2).toBeOnTheScreen();
  const welcomecardsareaImgElement = within(
    welcomeCardsWCAElement2,
  ).getByTestId('welcomecardsarea-logo');
  expect(welcomecardsareaImgElement).toBeOnTheScreen();
});

test('should render second welcomecardsarea: button with correct style ', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsWCAElement2 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea2',
  );
  expect(welcomeCardsWCAElement2).toBeOnTheScreen();
  const welcomecardsareaButtonElement = within(
    welcomeCardsWCAElement2,
  ).getByTestId('welcomecardsarea-button');
  expect(welcomecardsareaButtonElement).toBeOnTheScreen();
  expect(welcomecardsareaButtonElement).toHaveStyle({
    backgroundColor: secondary,
  });
  const buttonText = within(welcomecardsareaButtonElement).getByTestId(
    'button-text',
  );
  expect(buttonText).toHaveStyle({
    color: primary,
  });
});

test('should render second welcomecardsarea: should handle onButtonPress ', () => {
  render(<WelcomeCards {...props} />);
  const welcomecardsElement = screen.getByTestId('test-welcomecards-id');
  expect(welcomecardsElement).toBeOnTheScreen();
  const welcomeCardsWCAElement2 = within(welcomecardsElement).getByTestId(
    'welcomecards-welcomecardsarea2',
  );
  expect(welcomeCardsWCAElement2).toBeOnTheScreen();
  const welcomecardsareaButtonElement = within(
    welcomeCardsWCAElement2,
  ).getByTestId('welcomecardsarea-button');
  expect(welcomecardsareaButtonElement).toBeOnTheScreen();
  expect(handlePress).toBeCalledTimes(1);
  fireEvent.press(welcomecardsareaButtonElement);
  expect(handlePress).toBeCalledTimes(2);
});
