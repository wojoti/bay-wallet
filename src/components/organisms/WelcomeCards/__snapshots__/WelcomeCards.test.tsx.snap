// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "100%",
      },
      {
        "backgroundColor": "#363062",
      },
    ]
  }
  testID="test-welcomecards-id"
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": 1.2,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": "100%",
        },
        {
          "backgroundColor": "#363062",
        },
      ]
    }
  >
    <SvgMock
      height={330}
      testID="welcomecards-img"
      width="100%"
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "backgroundColor": "white",
          "borderColor": "transparent",
          "borderTopLeftRadius": 45,
          "borderTopRightRadius": 45,
          "borderWidth": 0,
        },
      ]
    }
    testID="welcomecards-radius"
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "backgroundColor": "transparent",
            "flex": 1,
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <RCTScrollView
        activeDotColor="#363062"
        automaticallyAdjustContentInsets={false}
        autoplay={false}
        autoplayDirection={true}
        autoplayTimeout={2.5}
        bounces={false}
        contentContainerStyle={
          [
            {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        contentOffset={
          {
            "x": 0,
          }
        }
        disableNextButton={false}
        disablePrevButton={false}
        horizontal={true}
        index={0}
        loadMinimal={false}
        loadMinimalSize={1}
        loop={false}
        onIndexChanged={[Function]}
        onMomentumScrollEnd={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        paginationStyle={
          {
            "bottom": 80,
            "justifyContent": "flex-start",
            "left": 28,
          }
        }
        pagingEnabled={true}
        removeClippedSubviews={true}
        scrollsToTop={false}
        showsButtons={false}
        showsHorizontalScrollIndicator={false}
        showsPagination={true}
        showsVerticalScrollIndicator={false}
      >
        <View>
          <View
            style={
              [
                {
                  "height": 1334,
                  "width": 750,
                },
                {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "stretch",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "flex-start",
                    "width": "100%",
                  },
                  {
                    "alignItems": "center",
                    "width": "100%",
                  },
                ]
              }
              testID="welcomecards-welcomecardsarea1"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "stretch",
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "flex-start",
                      "width": "100%",
                    },
                    {
                      "marginHorizontal": undefined,
                      "marginTop": undefined,
                      "width": "85%",
                    },
                  ]
                }
                testID="welcomecardsarea-card"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "stretch",
                        "flex": undefined,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "width": "100%",
                      },
                      {
                        "marginHorizontal": undefined,
                        "marginTop": 50,
                        "width": undefined,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "stretch",
                          "flex": undefined,
                          "flexDirection": "column",
                          "justifyContent": "flex-start",
                          "width": "100%",
                        },
                        {
                          "marginHorizontal": undefined,
                          "marginTop": undefined,
                          "width": "78%",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "black",
                          "fontFamily": "Poppins-Bold",
                          "fontSize": 40,
                          "textAlign": "left",
                        }
                      }
                      testID="welcomecardsarea-header"
                    >
                      maintext1
                    </Text>
                  </View>
                  <View
                    style={
                      [
                        {
                          "alignItems": "stretch",
                          "flex": undefined,
                          "flexDirection": "column",
                          "justifyContent": "flex-start",
                          "width": "100%",
                        },
                        {
                          "marginHorizontal": undefined,
                          "marginTop": undefined,
                          "width": "auto",
                        },
                      ]
                    }
                  >
                    <SvgMock
                      height={60}
                      testID="welcomecardsarea-logo"
                      width={60}
                    />
                  </View>
                </View>
                <View
                  style={
                    [
                      {
                        "alignItems": "stretch",
                        "flex": undefined,
                        "flexDirection": "column",
                        "justifyContent": "flex-start",
                        "width": "100%",
                      },
                      {
                        "marginHorizontal": undefined,
                        "marginTop": 20,
                        "width": undefined,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      {
                        "color": "black",
                        "fontFamily": "Poppins-Light",
                        "fontSize": 14,
                        "fontWeight": "400",
                        "textAlign": "left",
                      }
                    }
                    testID="welcomecardsarea-label"
                  >
                    cardtext1
                  </Text>
                </View>
                <View
                  style={
                    [
                      {
                        "alignItems": "flex-end",
                        "flex": undefined,
                        "flexDirection": "column",
                        "justifyContent": "flex-start",
                        "width": "100%",
                      },
                      {
                        "marginHorizontal": undefined,
                        "marginTop": 30,
                        "width": undefined,
                      },
                    ]
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "white",
                        "borderColor": "white",
                        "borderRadius": 8,
                        "borderStyle": "solid",
                        "height": 65,
                        "justifyContent": "center",
                        "opacity": 1,
                      }
                    }
                    testID="welcomecardsarea-button"
                  >
                    <Text
                      style={
                        {
                          "color": "#363062",
                          "fontFamily": "Poppins-Bold",
                          "fontSize": 18,
                          "textAlign": "center",
                        }
                      }
                      testID="button-text"
                    >
                      GET STARTED
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              [
                {
                  "height": 1334,
                  "width": 750,
                },
                {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "stretch",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "flex-start",
                    "width": "100%",
                  },
                  {
                    "alignItems": "center",
                    "width": "100%",
                  },
                ]
              }
              testID="welcomecards-welcomecardsarea2"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "stretch",
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "flex-start",
                      "width": "100%",
                    },
                    {
                      "marginHorizontal": undefined,
                      "marginTop": undefined,
                      "width": "85%",
                    },
                  ]
                }
                testID="welcomecardsarea-card"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "stretch",
                        "flex": undefined,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "width": "100%",
                      },
                      {
                        "marginHorizontal": undefined,
                        "marginTop": 50,
                        "width": undefined,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "stretch",
                          "flex": undefined,
                          "flexDirection": "column",
                          "justifyContent": "flex-start",
                          "width": "100%",
                        },
                        {
                          "marginHorizontal": undefined,
                          "marginTop": undefined,
                          "width": "78%",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "black",
                          "fontFamily": "Poppins-Bold",
                          "fontSize": 40,
                          "textAlign": "left",
                        }
                      }
                      testID="welcomecardsarea-header"
                    >
                      maintext2
                    </Text>
                  </View>
                  <View
                    style={
                      [
                        {
                          "alignItems": "stretch",
                          "flex": undefined,
                          "flexDirection": "column",
                          "justifyContent": "flex-start",
                          "width": "100%",
                        },
                        {
                          "marginHorizontal": undefined,
                          "marginTop": undefined,
                          "width": "auto",
                        },
                      ]
                    }
                  >
                    <SvgMock
                      height={60}
                      testID="welcomecardsarea-logo"
                      width={60}
                    />
                  </View>
                </View>
                <View
                  style={
                    [
                      {
                        "alignItems": "stretch",
                        "flex": undefined,
                        "flexDirection": "column",
                        "justifyContent": "flex-start",
                        "width": "100%",
                      },
                      {
                        "marginHorizontal": undefined,
                        "marginTop": 20,
                        "width": undefined,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      {
                        "color": "black",
                        "fontFamily": "Poppins-Light",
                        "fontSize": 14,
                        "fontWeight": "400",
                        "textAlign": "left",
                      }
                    }
                    testID="welcomecardsarea-label"
                  >
                    cardtext2
                  </Text>
                </View>
                <View
                  style={
                    [
                      {
                        "alignItems": "flex-end",
                        "flex": undefined,
                        "flexDirection": "column",
                        "justifyContent": "flex-start",
                        "width": "100%",
                      },
                      {
                        "marginHorizontal": undefined,
                        "marginTop": 30,
                        "width": undefined,
                      },
                    ]
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "white",
                        "borderColor": "white",
                        "borderRadius": 8,
                        "borderStyle": "solid",
                        "height": 65,
                        "justifyContent": "center",
                        "opacity": 1,
                      }
                    }
                    testID="welcomecardsarea-button"
                  >
                    <Text
                      style={
                        {
                          "color": "#363062",
                          "fontFamily": "Poppins-Bold",
                          "fontSize": 18,
                          "textAlign": "center",
                        }
                      }
                      testID="button-text"
                    >
                      GET STARTED
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        pointerEvents="none"
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "bottom": 25,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
            },
            {
              "bottom": 80,
              "justifyContent": "flex-start",
              "left": 28,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "#363062",
                "borderRadius": 4,
                "height": 8,
                "marginBottom": 3,
                "marginLeft": 3,
                "marginRight": 3,
                "marginTop": 3,
                "width": 8,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            [
              {
                "backgroundColor": "rgba(0,0,0,.2)",
                "borderRadius": 4,
                "height": 8,
                "marginBottom": 3,
                "marginLeft": 3,
                "marginRight": 3,
                "marginTop": 3,
                "width": 8,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
